### PR TITLE
Docker builds + push to AWS on master merge

### PR DESCRIPTION
**Pull request by nucleargriff**
TL;DR - after this merges, each merge to master should auto-push to https://lichess-staging.primer.com

### Changelog

1. Updates `build.sbt` with overrides for sbt's built-in Universal packager for Docker images to allow for a custom entry point & additional Dockerfile config. We could technically avoid the overrides by making some larger changes to logging strategy, but being able to add additional entries to the generated Dockerfile is likely to be useful down the road.
2. Creates `prod.conf` with ENV var driven configuration. These ENV vars are injected at deploy time into the container at ECS side. Technically this could be a smaller config with overrides from `base.conf`, but I have a feeling in time the configs will diverge enough where a separate config is warranted. Can iterate on this though.
3. Creates a github workflow that automatically builds all assets (read: UI), compiles the docker image (tagging it with `latest`), pushes to ECR (Amazon's private docker registry), and informs the ECS cluster service to re-deploy. Eventually we'll want to be tagging the docker images with the specific commit so roll backs are easier, but relying on latest should be okay for now.

### Testing
I've tested all this manually & the image built from this branch is currently running in staging. Not much for others to test pre-merge, but you _can_ run the docker build locally and see what comes of it: `sbt docker:publishLocal`.

Follow up PR to `lichess-dev` repo coming soon that will enable a full-stack "prod-like" docker-compose.

### CODE REVIEW

**Comment by bjoveski** [comment]
woo, very cool!, feel free to merge!
